powersimdata/input/check.py

```diff
@@ -299,19 +299,23 @@ def _check_grid_type(grid):
         raise TypeError(f"grid must be a {_grid.Grid} object")
 
 
-def _check_areas_and_format(areas, grid_model="usa_tamu"):
+def _check_areas_and_format(areas, mi=ModelImmutables("usa_tamu")):
     """Ensure that areas are valid. Duplicates are removed and state abbreviations are
     converted to their actual name.
 
     :param str/list/tuple/set areas: areas(s) to check. Could be load zone name(s),
         state name(s)/abbreviation(s) or interconnect(s).
-    :param str grid_model: grid model.
-    :raises TypeError: if areas is not a list/tuple/set of str.
-    :raises ValueError: if areas is empty or not valid.
+    :param powersimdata.network.model.ModelImmutables mi: immutables of a grid model.
+    :param int reduction: reduction parameter (number of nodes in network). If None,
+        the full network is loaded. Applies to the *'europe_tub'* grid model.
+    :raises TypeError: if ``areas`` is not a list/tuple/set of str.
+    :raises ValueError:
+        if ``areas`` is empty or not valid.
     :return: (*set*) -- areas as a set. State abbreviations are converted to state
         names.
     """
-    mi = ModelImmutables(grid_model)
+    division = [a for a in mi.zones["mappings"] if "abbr" in a][0].split("_")[0]
```

Comment:
I agree, it is a quick and dirty to distinguish US models from European model (state vs country). I will work on it.

Reply:
@BainanXia and @danielolsen, can you look at 56d1a464489d3e17d59717e5085d9aeebbe4c4c7 and tell me if you like the implementation?

Reply:
It should work for the current use case, where we have exactly three items in the list `["interconnect", SOMETHING, "loadzone"]`. I'm not sure how it would extend if we wanted to be able to add sub-country divisions, but maybe that's far beyond what we need right now. Either way it's definitely cleaner than the list comprehension solution.

Reply:
I thought about that. We will have to augment the hierarchy with new areas (counties, province) and decide which area we want to map to/from loadzone.

Reply:
I kept things simple and added a `division` key in the `zones` attribute of a `ModelImmutables` object with value set to `state` or `country` depending on the grid model. This will do the job for now.